precompiles/proxy/Proxy.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.0;
+
+/**
+ * @author The Moonbeam Team
+ * @title The interface through which solidity contracts will interact with the Proxy pallet
+ * We follow this same interface including four-byte function selectors, in the precompile that
+ * wraps the pallet
+ * Address :    0x000000000000000000000000000000000000080a
+ */
+
+interface Proxy {
+    /// @dev Defines the proxy permission types that may be combined via `|` operator
+    /// The values start at `0` and are represented as `uint32`
+    enum ProxyType {
+        Any,
+        NonTransfer,
+        Governance,
+        Staking,
+        CancelProxy,
+        Balances,
+        AuthorMapping,
+        IdentityJudgement
+    }
+
+    /// @dev Dispatch the given call from an account that the sender is authorised for through
+    /// addProxy. Removes any corresponding announcement(s)
+    /// @param real the account that the proxy will make a call on behalf of
+    /// @param forceProxyType the exact proxy type to be used and checked for this call, encoded 
+    /// as bytes[] where first byte is either 0x00 for not using the parameter, or `0x01` followed
+    /// by big-endian encoded `uint32` ProxyType
+    /// @param call the call to be made by the real account
+    /// @return A boolean confirming whether the address is a nominator
+    /// Selector: a
+    function proxy(
+        address real, 
+        bytes[] forceProxyType,
```

Comment:
Maybe we can add an enum variant being "NoProxy", which would indicate we need to insert a None here. Would that look better than having bytes[]?

Reply:
Maybe easier to understand for users

Reply:
Nah forget it, just checked again and this is in case you want to specifically force the proxy to something.

Reply:
One thing we can do is offer two precompile fucntions, `proxy_with_requirements` (I am nto good at names) and simply `proxy`. The first would pass a Some(ProxyType) and the latter would pass None

Reply:
I agree having a different function seems more idiomatic, perhaps I will cross check with Alberto on how SC users normally prefer their function parameters, and update it. Thanks.